lib/workload/stateless/universal-event-archiver/deploy/component.ts

```diff
@@ -0,0 +1,74 @@
+import path from 'path';
+import * as cdk from 'aws-cdk-lib';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { aws_lambda, Duration, Stack } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { PythonFunction } from '@aws-cdk/aws-lambda-python-alpha';
+import { Architecture } from 'aws-cdk-lib/aws-lambda';
+import { ISecurityGroup, IVpc } from 'aws-cdk-lib/aws-ec2';
+import { IEventBus, Rule } from 'aws-cdk-lib/aws-events';
+
+export interface UniversalEventArchiverProps {
+  securityGroups: ISecurityGroup[];
+  vpc: IVpc;
+  archiveBucket: IBucket;
+  mainBus: IEventBus;
```

Comment:
For vpc, bucket and mainBus here, should take string type of their names. Then, later in the constructor, you should do from lookup to reconstruct them into their construct object form. Pls leverage `vpcProps`, `eventBusName` from `config/constants.ts`. And, create `eventArchiveBucketName` similarly and, pass-in to stack constructor.

Reply:
Ah (correction). `vpcProps` is no more simple string nor object. But `VpcLookupOptions` type.

Reply:
Eh. I commented base on down under extending the Stack. 

Whether refactoring as a construct into event bus stack... Umm. That would also do, perhaps.

Reply:
yes, we can get these objects in the stack level when combined it stateless stack. I will create the universalEventArchiverConfig if needed  and combine with the mainbus and vpc we already have in the stack.

Reply:
You don't want to keep this as independent stack as it is? May be keeping it as independent stack for strategic starter...

Reply:
Why independent? Would we ever want to manage this independent of the event bus?
We can make it optional in the event bus stack (or even the event bus construct).

Reply:
Um. Just don't want to have side effect to core infra. This is new and there are few unknown variable yet. But, sure, go for it. 

I reckon, I might be over protective. So, please go for it as refactoring as construct into shared stateful stack. Perhaps that's ok.

Reply:
I'd add it to the construct and make it optional controlled by a boolean flag (like `addCustomEventArchive`).
The construct would check whether required props are present (e.g. the output bucket) and the jest tests could test for both options. 

That way it should not change anything on the existing infra when the flag is off. We can run a test deploy to check and then switch it on when feel confident. 
It should also allow us to remove it if/when we want without affecting the bus deployment.

Reply:
Thanks, I will refactor it into construct of shared stateful stack first

Reply:
Ok 👌 that sound like the plan.